internal/services/eventgrid/eventgrid_partner_namespace_resource_test.go

```diff
@@ -0,0 +1,223 @@
+		},
+		{
+			Config:      r.requiresImport(data),
+			ExpectError: acceptance.RequiresImportError("azurerm_eventgrid_partner_namespace"),
+		},
+	})
+}
+
+func TestAccEventGridPartnerNamespace_complete(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_eventgrid_partner_namespace", "test")
+	r := EventGridPartnerNamespaceTestResource{}
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.complete(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccEventGridPartnerNamespace_update(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_eventgrid_partner_namespace", "test")
+	r := EventGridPartnerNamespaceTestResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.complete(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.completeUpdate(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func (EventGridPartnerNamespaceTestResource) Exists(ctx context.Context, client *clients.Client, state *pluginsdk.InstanceState) (*bool, error) {
+	id, err := partnernamespaces.ParsePartnerNamespaceID(state.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := client.EventGrid.PartnerNamespaces.Get(ctx, *id)
+	if err != nil {
+		return nil, fmt.Errorf(("retrieving %s: %+v"), *id, err)
+	}
+
+	return pointer.To(resp.Model != nil), nil
+}
+
+func (r EventGridPartnerNamespaceTestResource) basic(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctest-rg-%[1]d"
+  location = "%[2]s"
+}
+
+resource "azurerm_eventgrid_partner_registration" "test" {
+  name                = "acctest-egpr-%[1]d"
+  resource_group_name = azurerm_resource_group.test.name
+}
+
+resource "azurerm_eventgrid_partner_namespace" "test" {
+  name                    = "acctest-egpn-%[1]d"
+  location                = "%[2]s"
+  resource_group_name     = azurerm_resource_group.test.name
+  partner_registration_id = azurerm_eventgrid_partner_registration.test.id
+}
+`, data.RandomInteger, data.Locations.Primary)
+}
+
+func (EventGridPartnerNamespaceTestResource) requiresImport(data acceptance.TestData) string {
+	template := EventGridPartnerNamespaceTestResource{}.basic(data)
+
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_eventgrid_partner_namespace" "import" {
+  name                    = azurerm_eventgrid_partner_namespace.test.name
+  location                = azurerm_eventgrid_partner_namespace.test.location
+  resource_group_name     = azurerm_eventgrid_partner_namespace.test.resource_group_name
+  partner_registration_id = azurerm_eventgrid_partner_namespace.test.partner_registration_id
+}
+`, template)
```

Comment:
We can pass this as an argument directly to the fmt.Sprintf function

```suggestion
	return fmt.Sprintf(`
%s

resource "azurerm_eventgrid_partner_namespace" "import" {
  name                    = azurerm_eventgrid_partner_namespace.test.name
  location                = azurerm_eventgrid_partner_namespace.test.location
  resource_group_name     = azurerm_eventgrid_partner_namespace.test.resource_group_name
  partner_registration_id = azurerm_eventgrid_partner_namespace.test.partner_registration_id
}
`, EventGridPartnerNamespaceTestResource{}.basic(data))
```